### PR TITLE
Add using documentation for SageMaker

### LINKED ISSUE #1814: Sagemaker with lakeFS documentation
sagemaker is aws leading platform for ai, it works with lakeFS so let's document it.

### PR DESCRIPTION
**Pull request by itaiad200**
closes #1814

### CODE REVIEW

File: docs/using/sagemaker.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+## Table of contents
+{: .no_toc .text-delta }
+
+1. TOC
+{:toc}
+
+## Initializing session and client
+
+Initialize sagemaker session and s3 client with lakeFS as the endpoint:
```
Comment (johnnyaug) [inline docs/using/sagemaker.md:22]:
```suggestion
Initialize a SageMaker session and an S3 client with lakeFS as the endpoint:
```

File: docs/using/sagemaker.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+```python
+import sagemaker
+import boto3
+
+endpoint_url='<LAKEFS_S3_GATEWAY_ENDPOINT>'
+aws_access_key_id='<LAKEFS_ACCESS_KEY_ID>'
+aws_secret_access_key='<LAKEFS_SECRET_ACCESS_KEY>'
```
Comment (johnnyaug) [inline docs/using/sagemaker.md:29]:
```suggestion
endpoint_url = '<LAKEFS_S3_GATEWAY_ENDPOINT>'
aws_access_key_id = '<LAKEFS_ACCESS_KEY_ID>'
aws_secret_access_key = '<LAKEFS_SECRET_ACCESS_KEY>'
```

File: docs/using/sagemaker.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+### Download objects
+
+We can use the integration with lakeFS to download a portion of the data we see fit:
+ 
+```python
+repo = 'example-repo'
+prefix = "/prefix-to-download"
+branch = 'main'
+
+session.download_data(path='./'+branch, bucket=repo, key_prefix = branch + prefix)
```
Comment (johnnyaug) [inline docs/using/sagemaker.md:74]: the path here is confusing, maybe put  in variable called local_path or something

Comment (itaiad200) [reply docs/using/sagemaker.md]: Done.

**Review by johnnyaug — APPROVED** [review]

**Pull request merged**